test/widgets/set_status_test.dart

```diff
@@ -0,0 +1,458 @@
+import 'dart:io';
+
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_checks/flutter_checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart' as http;
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/route/realm.dart';
+import 'package:zulip/basic.dart';
+import 'package:zulip/model/emoji.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/button.dart';
+import 'package:zulip/widgets/emoji_reaction.dart';
+import 'package:zulip/widgets/icons.dart';
+import 'package:zulip/widgets/page.dart';
+import 'package:zulip/widgets/profile.dart';
+import 'package:zulip/widgets/set_status.dart';
+import 'package:zulip/widgets/user.dart';
+
+import '../api/fake_api.dart';
+import '../example_data.dart' as eg;
+import '../example_data.dart';
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import '../stdlib_checks.dart';
+import '../test_images.dart';
+import '../test_navigation.dart';
+
+import 'checks.dart';
+import 'finders.dart';
+import 'test_app.dart';
+
+void main() {
+  late PerAccountStore store;
+
+  TestZulipBinding.ensureInitialized();
+
+  final Map<String, List<String>> suggestedUnicodeEmoji = {
+    '1f6e0': ['working_on_it'],
+    '1f4c5': ['calendar'],
+    '1f68c': ['bus'],
+    '1f912': ['sick'],
+    '1f334': ['palm_tree'],
+    '1f3e0': ['house'],
+    '1f3e2': ['office'],
+  };
+  final ServerEmojiData suggestedEmojiData = ServerEmojiData(codeToNames: suggestedUnicodeEmoji);
+
+  Future<void> setupPage(WidgetTester tester, {
+    UserStatusChange change = const UserStatusChange(text: OptionNone(), emoji: OptionNone()),
+    ServerEmojiData? emojiData,
+    NavigatorObserver? navigatorObserver,
+  }) async {
+    addTearDown(testBinding.reset);
+
+    Route<dynamic>? currentRoute;
+    final testNavObserver = TestNavigatorObserver()
+      ..onPushed = (route, _) => currentRoute = route;
+
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+    await store.addUser(eg.selfUser);
+    if (change.text != OptionNone<String?>() || change.emoji != OptionNone<StatusEmoji?>()) {
+      await store.changeUserStatus(eg.selfUser.userId, change);
```

Comment:
nit: can make this simpler:
```suggestion
    if (change != null) {
      await store.changeUserStatus(eg.selfUser.userId, change!);
```

and let `change` default to null.